Firestore/core/src/firebase/firestore/api/collection_reference.mm

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/api/collection_reference.h"
+
+#include <utility>
+
+#import "Firestore/Source/Core/FSTQuery.h"
+
+#include "Firestore/core/src/firebase/firestore/api/document_reference.h"
+#include "Firestore/core/src/firebase/firestore/api/input_validation.h"
+#include "Firestore/core/src/firebase/firestore/model/document_key.h"
+#include "Firestore/core/src/firebase/firestore/model/resource_path.h"
+#include "Firestore/core/src/firebase/firestore/util/autoid.h"
+#include "Firestore/core/src/firebase/firestore/util/hashing.h"
+#include "Firestore/core/src/firebase/firestore/util/string_apple.h"
+
+namespace util = firebase::firestore::util;
+using firebase::firestore::api::ThrowInvalidArgument;
+using firebase::firestore::model::DocumentKey;
+using firebase::firestore::model::ResourcePath;
+
+namespace firebase {
+namespace firestore {
+namespace api {
+
+namespace {
+
+FSTQuery* MakeQuery(model::ResourcePath path) {
+  if (path.size() % 2 != 1) {
+    ThrowInvalidArgument("Invalid collection reference. Collection references "
+                         "must have an odd number of segments, but %s has %s",
```

Comment:
This message is synced across platforms and tested for an exact match. Since we're not doing this for other similar messages I don't want to introduce the inconsistency.

If you'd like to raise this past the point of an ultranit, let's discuss at API discussion or in chat.

Reply:
No, changing this across platforms and messages certainly doesn't seem like something to do at the moment.